students/alex_whitty/lesson3/mailroom.py

```diff
@@ -0,0 +1,106 @@
+import sys  # imports go at the top of the file
+
+
+# Data Structure
+donor_db = {"William Gates, III": [653772.32, 12.17],
+            "Jeff Bezos": [877.33],
+            "Paul Allen": [663.23, 43.87, 1.32],
+            "Mark Zuckerberg": [1663.23, 4300.87, 10432.0],
+            "Elon Musk": [2263.23, 3300.87, 15432.0],
+            }
+
+
+
+# Input/Output
+prompt = "\n".join(("Thanking your donors for their generous gifts!",
+          "Please choose from below options:",
+          "1 - Send a Thank You",
+          "2 - Create a Report",
+          "3 - Exit",
+          ">>> "))
+
+# Features
+def thank_you():
+
+    answer = input("Please enter Full Name.>>>")
+    if answer.lower() == 'list':
+        for donor in donor_db.keys():
+            print(donor)
+    else:
+        if answer in donor_db:
+            current_donor(answer)
+        else:
+            add_donor()
+
+
+def current_donor(donor):
+
+    answer = input("Please enter donor Full Name.>>>")
+    if answer.title() in donor_db:
+        amount = input("Please enter donor amount.>>>")
+        donor_db[donor].append(float(amount))
+
+
+
+def sort_key(donor_db):
+    return donor_db[:]
+sorted(donor_db, key=sort_key)
+
+
+def add_donor():
+
+
+    new_donor = input("Name of the new donor to add?").title()
+    donor_amount = input("What is the donation amount?")
+    donor_db[new_donor] = [float(donor_amount)]
+    print(donor_db)
+    print('Thanks, {} for your generous donation.'.format(new_donor))
+
```

Comment:
_add_donor()_ should become _add_donor(new_donor)_, so that you don't need to ask for the name of the donor again.